tools/codegen/gen_functionalization_type.py

```diff
@@ -0,0 +1,426 @@
+    # The purpose of the translate is to get out expressions that convert the references to value types,
+    # so we can store them in a lambda capture.
+    non_self_capture_exprs = translate(non_self_bindings, non_self_value_ctypes, method=False)
+    captured_view_args_str = ', '.join(f'{val.type.name} = {val.expr}' for val in non_self_capture_exprs)
+    # All of the captured variables are now in scope inside of the lambda. Add them to the context.
+    lambda_capture_ctx = [Expr(expr=str(ctype.name), type=ctype) for ctype in non_self_value_ctypes]
+
+    forward_lambda_args, forward_name, forward_ret_type, forward_call_bindings = view_lambda_types(f, reverse=False)
+    reverse_lambda_args, reverse_name, reverse_ret_type, reverse_call_bindings = view_lambda_types(f, reverse=True)
+
+    forward_lambda_ctx: Sequence[Expr] = lambda_capture_ctx + [Expr(expr=b.name, type=b.nctype) for b in forward_lambda_args]
+    reverse_lambda_ctx: Sequence[Expr] = lambda_capture_ctx + [Expr(expr=b.name, type=b.nctype) for b in reverse_lambda_args]
+
+    forward_lambda_exprs = [e.expr for e in translate(forward_lambda_ctx, forward_call_bindings, method=False)]
+    reverse_lambda_exprs = [e.expr for e in translate(reverse_lambda_ctx, reverse_call_bindings, method=False)]
+
+    maybe_idx = ''
+    if is_multi_output(f.func.returns):
+        #  mutated_view_idx is only needed for lambdas of multi-output view ops, like split().
+        # For the forward lambda, we just directly call the corresponding view op and index into the output,
+        # which is why mutated_view_idx isn't actually an argument to the function call.
+        # For the reverse lambda, we have a calling convention for every {view}_reverse function such that
+        # it only accepts an additional mutated_view_idx parameter if it's a multi-output view op.
+        maybe_idx = f'[{forward_lambda_exprs[-1]}]'
+        forward_lambda_exprs = forward_lambda_exprs[:-1]
+
+    if Variant.method in f.variants:
+        forward_lambda_call_str = f'{forward_lambda_exprs[0]}.{forward_name}({", ".join(forward_lambda_exprs[1:])}){maybe_idx}'
+    else:
+        forward_lambda_call_str = f'at::{forward_name}({", ".join(forward_lambda_exprs)}){maybe_idx}'
+
+    view_meta_str = f"""
+      at::functionalization::ViewMeta view_meta = at::functionalization::ViewMeta(
+        [{captured_view_args_str}]({', '.join(a.decl() for a in forward_lambda_args)}) -> {forward_ret_type.cpp_type()} {{
+          return {forward_lambda_call_str};
+        }},
+        [{captured_view_args_str}]({', '.join(a.decl() for a in reverse_lambda_args)}) -> {reverse_ret_type.cpp_type()} {{
+          return at::functionalization::impl::{reverse_name}({", ".join(reverse_lambda_exprs)});
+        }}
+      );"""
+
+    api_name = CppSignatureGroup.from_native_function(
+        f, method=False, fallback_binding=f.manual_cpp_binding).most_faithful_signature().name()
```

Comment:
I guess I'm a little confused why you don't just use the `at::_ops` API here either, since that API was designed to be easy to codegen calls to. (Instead, you have futz about here to extract the cpp name of the function. But then you make the call using dispatch signature below! You're lucky they coincided in this case...)